Package updates

etungsten opened the pull request:
<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**
N/A


**Description of changes:**
Updates third-party dependencies in `packages/`.

Excluding ecs-agent as that is being covered by https://github.com/bottlerocket-os/bottlerocket/pull/1201
Excluding containerd 1.4 which is a major update that requires further inspection and testing.
systemd v247 and dbus-broker 25 update is in https://github.com/bottlerocket-os/bottlerocket/pull/1265

```
* 1d7baa43 - kubernetes-1.18: update to 1.18.14 <Erikson Tung>
* 3edc27cc - kubernetes-1.17: update to 1.17.13 <Erikson Tung>
* 2a7cdcf1 - docker-proxy: update to 19.03.14 <Erikson Tung>
* 2c8c9e72 - docker-engine: update to 19.03.14 <Erikson Tung>
* 5351e16a - docker-cli: update to 19.03.14 <Erikson Tung>
* 944d54b3 - tcpdump: update to 4.99.0 <Erikson Tung>
* c90749f7 - socat: update to 1.7.4.0 <Erikson Tung>
* d765c8c3 - readline: update to 8.1 <Erikson Tung>
* e432874e - libseccomp: update to 2.5.1 <Erikson Tung>
* 0289a32f - libpcre: update to 10.36 <Erikson Tung>
* d29c8f6e - libcap: update to 2.44 <Erikson Tung>
* b4b9c360 - libaudit: update to 3.0 <Erikson Tung>
* 65f4af13 - cni-plugins: update to 0.9.0 <Erikson Tung>
* 30c58e3e - ca-certificates: update to 2020.12.08 <Erikson Tung>
```


**Testing done:**

- [x] aws-k8s-1.15, aws-k8s-1.16, aws-k8s-1.17, aws-k8s-1.18, aws-k8s-1.19 on x86 and arm64
- Can run pods OK
- kubelet status look good, journal looks good
- journal in general looks normal

- [x] aws-ecs-1
- Runs tasks ok

- [x] aws-dev
- Runs containers ok


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

etungsten committed:
ca-certificates: update to 2020.12.08

etungsten committed:
cni-plugins: update to 0.9.0

etungsten committed:
libaudit: update to 3.0

etungsten committed:
libcap: update to 2.44

etungsten committed:
libpcre: update to 10.36

etungsten committed:
libseccomp: update to 2.5.1

etungsten committed:
readline: update to 8.1

etungsten committed:
socat: update to 1.7.4.0

etungsten committed:
tcpdump: update to 4.99.0

etungsten committed:
docker-cli: update to 19.03.14

etungsten committed:
docker-engine: update to 19.03.14

We also bump the version of docker-init to match docker.
The versioning is there for consistency and doesn't reflect upstream versions
of tini, so it should match the docker we now ship.

This also drops patch for updating aws-sdk-go for IMDSv2 support.
Upstream has updated aws-sdk-go to support IMDSv2

etungsten committed:
kubernetes-1.17: update to 1.17.13

etungsten committed:
kubernetes-1.18: update to 1.18.14

tjkirch commented on packages/docker-proxy/docker-proxy.spec:
```diff
@@ -8,7 +8,7 @@
 
 Name: %{_cross_os}docker-proxy
-Version: 19.03.13
+Version: 19.03.14
```
It won't let me comment on an unchanged line above, but I think we have to update the libnetwork version as well, any time we update docker.  https://github.com/moby/moby/blob/v19.03.14/vendor.conf shows commit 55e924b8a84231a065879156c0de95aefc5f5435, I believe, and above we're on an older 026aabaa.

etungsten replied:
Good catch

etungsten committed:
docker-proxy: update to 19.03.14

Bumps libnetwork to 55e924b8a84231a065879156c0de95aefc5f5435 to match
docker 19.03.14 release

etungsten commented:
Pushes above address https://github.com/bottlerocket-os/bottlerocket/pull/1264#discussion_r553563167

etungsten commented:
Push above rebases onto develop to pull in aws-k8s-1.19 variant.

zmrow reviewed (approved):
:bike:
